scripts/helper_scripts/unipept-database-rs/src/taxons_uniprots_tables/table_writer.rs

```diff
@@ -0,0 +1,226 @@
+    pub fn new(
+        taxons: &PathBuf,
+        peptides: &PathBuf,
+        uniprot_entries: &PathBuf,
+        go_references: &PathBuf,
+        ec_references: &PathBuf,
+        interpro_references: &PathBuf,
+    ) -> Result<Self> {
+        Ok(TableWriter {
+            taxons: parse_taxon_file_basic(taxons).context("Unable to parse taxonomy file")?,
+            wrong_ids: HashSet::new(),
+            peptides: open_write(peptides).context("Unable to open output file")?,
+            uniprot_entries: open_write(uniprot_entries).context("Unable to open output file")?,
+            go_cross_references: open_write(go_references).context("Unable to open output file")?,
+            ec_cross_references: open_write(ec_references).context("Unable to open output file")?,
+            ip_cross_references: open_write(interpro_references)
+                .context("Unable to open output file")?,
+
+            peptide_count: 0,
+            uniprot_count: 0,
+            go_count: 0,
+            ec_count: 0,
+            ip_count: 0,
+        })
+    }
+
+    // Store a complete entry in the database
+    pub fn store(&mut self, entry: Entry) -> Result<()> {
+        let id = self
+            .write_uniprot_entry(&entry)
+            .context("Failed to write Uniprot entry")?;
+
+        // Failed to add entry
+        if id == -1 {
+            return Ok(());
+        }
+
+        for r in &entry.go_references {
+            self.write_go_ref(r, id);
+        }
+
+        for r in &entry.ec_references {
+            self.write_ec_ref(r, id);
+        }
+
+        for r in &entry.ip_references {
+            self.write_ip_ref(r, id);
+        }
+
+        let go_ids = entry.go_references.into_iter();
+        let ec_ids = entry
+            .ec_references
+            .iter()
+            .filter(|x| !x.is_empty())
+            .map(|x| format!("EC:{}", x));
+        let ip_ids = entry
+            .ip_references
+            .iter()
+            .filter(|x| !x.is_empty())
+            .map(|x| format!("IPR:{}", x));
+
+        let summary = go_ids
+            .chain(ec_ids)
+            .chain(ip_ids)
+            .collect::<Vec<String>>()
+            .join(";");
+
+        for sequence in calculate_entry_digest(
+            &entry.sequence,
+            entry.min_length as usize,
+            entry.max_length as usize,
+        ) {
+            self.write_peptide(
+                sequence
+                    .iter()
+                    .map(|&x| if x == b'I' { b'L' } else { x })
+                    .collect(),
+                id,
+                sequence,
+                &summary,
+            )
+            .context("Failed to write peptide")?;
+        }
+
+        Ok(())
+    }
+
+    fn write_peptide(
+        &mut self,
+        sequence: Vec<u8>,
+        id: i64,
+        original_sequence: &[u8],
+        annotations: &String,
+    ) -> Result<()> {
+        self.peptide_count += 1;
+
+        if let Err(e) = writeln!(
```

Comment:
This is probably a fatal error as well, so returning an `Err` (by using the `?`-syntax) is best here I think?

Reply:
@bmesuere I think we should maybe let the application crash when an error happens here? (Instead of only logging the error). Right now there's the possibility that we don't spot an error and that a possibly incomplete database is being used. What's your opinion?

Reply:
This indeed seems fatal to me, so ok to quit.